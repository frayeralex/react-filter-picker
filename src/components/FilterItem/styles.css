.FilterItem {
  border: 1px solid #000;
  padding: 5px;
  margin: 2px;
  cursor: pointer;
}

.FilterItem_active {
  background: rgba(0, 0,0, .1);
}
