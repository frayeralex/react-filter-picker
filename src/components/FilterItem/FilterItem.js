import React from 'react';
import classNames from 'classnames';
import styles from './styles.css';
import PropTypes from 'prop-types';

const FilterItem = props => {
  return (
    <div
      onClick={props.onClick}
      className={classNames(styles.FilterItem, {
        [styles.FilterItem_active]: props.active,
      })}
    >
      <div className="title">{props.title}</div>
    </div>
  );
};

FilterItem.propTypes = {
  title: PropTypes.string.isRequired,
  active: PropTypes.bool,
  onClick: PropTypes.func,
};

FilterItem.defaultProps = {
  active: false,
  onClick: () => {},
};

export default FilterItem;
