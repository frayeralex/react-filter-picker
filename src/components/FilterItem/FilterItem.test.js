import React from 'react';
import FilterItem from './FilterItem';

const props = {
  title: 'title',
};

describe('<FilterItem/>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<FilterItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should has active mode', () => {
    const wrapper = shallow(<FilterItem {...props} />);
    wrapper.setProps({ active: true });
    expect(wrapper).toMatchSnapshot();
  });
  it('should correct handle onClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<FilterItem {...props} onClick={onClick} />);
    wrapper.root().simulate('click', {});
    expect(onClick).toHaveBeenCalled();
  });
});
