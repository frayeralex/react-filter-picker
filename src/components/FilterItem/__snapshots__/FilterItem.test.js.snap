// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterItem/> should has active mode 1`] = `
<div
  className="undefined"
  onClick={[Function]}
>
  <div
    className="title"
  >
    title
  </div>
</div>
`;

exports[`<FilterItem/> should render without crashing 1`] = `
<div
  className=""
  onClick={[Function]}
>
  <div
    className="title"
  >
    title
  </div>
</div>
`;
